src/roles/firewall_service/tasks/main.yml

```diff
@@ -34,6 +34,5 @@
 
   when:
     - ansible_os_family == 'Debian'
-    - firewall_service is defined
+    - "{{ firewall_service }} is defined"
```

Comment:
maybe it touches on a bug in ansible? there is another variable firewall_service_port (iirc) and quoting here fixed an 'unknown variable' error for me.

Reply:
for the record: here's the error I received (multiple times)

> - name: "Ensure UFW service {{ firewall_service }} (port {{ ufw_service_ports['{{ firewall_service }}'] }}) allowed for list of servers (Debian only)" 
>  ^ here 
> We could be wrong, but this one looks like it might be an issue with 
> missing quotes.  Always quote template expression brackets when they 
> start a value. For instance: 
> 
>    with_items: 
>      - {{ foo }} 
> 
> Should be written as: 
> 
>    with_items: 
>      - "{{ foo }}"

The error was fixed after enclosing all of line 37 in double quotes with **firewall_service** in curly braces.  I know this quoting is not used on line 36, but for whatever reason that's what works on line 37.